### PR TITLE
Use C# 6.0 expression bodied properties

### PR DESCRIPTION
**Pull request by stil**
Since we're already using VS2015 to compile, using this C# 6.0 feature
would make codebase slightly lighter and more readable.

### CODE REVIEW

File: SteamKit2/SteamKit2/Base/ClientMsg.cs [hunk 1]
```diff
@@ -28,14 +28,15 @@ internal ClientMsgProtobuf( int payloadReserve = 0 )
...
+        public override bool IsProto => true;
+
         /// <summary>
         /// Gets the network message type of this client message.
         /// </summary>
         /// <value>
         /// The network message type.
         /// </value>
-        public override EMsg MsgType { get { return Header.Msg; } }
+        public override EMsg MsgType => Header.Msg;
```
Comment (VoiDeD) [inline SteamKit2/SteamKit2/Base/ClientMsg.cs]: I think an expression body change like this modifies the property to be both get and set, something we don't want here. Ditto for a bunch of get only properties.

Comment (Netshroud) [inline SteamKit2/SteamKit2/Base/ClientMsg.cs]: Expression-bodied properties are read-only, there's no setter.

Comment (VoiDeD) [inline SteamKit2/SteamKit2/Base/ClientMsg.cs]: Okay I see, I had also seen the syntax that involved `{ get; } = ...` and thought that was the method to correctly make a read only property.

Comment (WildCard65) [inline SteamKit2/SteamKit2/Base/ClientMsg.cs]: No the { get; } = thing is the same as the doing [DefaultValue()]

Comment (Netshroud) [inline SteamKit2/SteamKit2/Base/ClientMsg.cs]: This is patently false.

`{ get; } = ...` instructs the compiler to emit an assignment to the backing field in the class's constructor (or static constructor for a static property).

`DefaultValueAttribute` does nothing, a framework has to explicitly search for it for it to be of use.

Comment (WildCard65) [inline SteamKit2/SteamKit2/Base/ClientMsg.cs]: good to know then

**Comment by VoiDeD** [comment]
Alright so I think the bottom line is that it's probably not harmful to update this syntax, but we'd be changing syntax (and making our blame history slightly more convoluted) just for the sake of changing syntax, rather than fixing any underlying issues.

**Pull request closed without merging**